pkg/controllers/instances/installation_equals.go

```diff
@@ -0,0 +1,52 @@
+// SPDX-FileCopyrightText: 2023 "SAP SE or an SAP affiliate company and Gardener contributors"
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package instances
+
+import (
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/util/json"
+
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+)
+
+// InstallationSpecDeepEquals test whether two installation specs are deeply equal.
+func InstallationSpecDeepEquals(specA, specB *lsv1alpha1.InstallationSpec) bool {
```

Comment:
Not sure but why don't you check the size of the data mappings first and if both are equal you compare them one by one? Or do I miss here something?

Reply:
You are right. That makes perfectly sense. I will change taht.